root_profiler/engine/db.php

```diff
@@ -187,15 +154,13 @@ function freeresult($query_id = 0)
 
 	function error()
 	{
-		$result['message'] = mysql_error($this->db_connect_id);
-		$result['code'] = mysql_errno($this->db_connect_id);
+		$result['message'] = $this->dbh->errorCode();
+		$result['code'] = $this->dbh->errorInfo;
 
 		return $result;
 	}
 
-  } // class sql_db
-
-  $rpgDB = new mysql_db($DB_HOST, $DB_USER, $DB_PWD, $DB) or
-    __printFatalErr(mysql_error().'Failed to find database.', __LINE__, __FILE__);
+  } // class Database 
 
+  $rpgDB = new Database($DB_DSN, $DB_USER, $DB_PWD)
```

Comment:
No error on fail ?  Or is that automagic in new ?

Reply:
If the call to create a PDO object in the Database constructor fails, it will just throw an Exception. Since there is no exception handling anywhere, it will just r0uslt in the whole thing going down, which in turn will result in an error 500. This is fine as in such a situation, there isn't much that can be done anyway.